jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -668,16 +849,81 @@ ecma_op_typedarray_set_index_prop (ecma_object_t *obj_p,/**< a TypedArray object
     lit_magic_string_id_t class_id = ecma_object_get_class_name (obj_p);
     lit_utf8_byte_t *target_p = ecma_arraybuffer_get_buffer (arraybuffer_p) + byte_pos;
 
+#define TYPEDARRAY_SET_INDEXED_ELEMENT(type) \
+  int64_t v = (int64_t) value_num; \
+  *((type *) target_p) = (type) v;
+
     switch (class_id)
     {
       case LIT_MAGIC_STRING_INT8_ARRAY_UL:
       {
-        *((int8_t *) target_p) = (int8_t) value_num;
+        TYPEDARRAY_SET_INDEXED_ELEMENT (int8_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_UINT8_ARRAY_UL:
+      {
+        TYPEDARRAY_SET_INDEXED_ELEMENT (uint8_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_UINT8_CLAMPED_ARRAY_UL:
+      {
+        uint8_t clamped;
+        int64_t tmp_int;
+
+        if (value_num > 255)
+        {
+          clamped = 255;
+        }
+        else if (value_num < 0)
+        {
+          clamped = 0;
+        }
+        else
+        {
+          tmp_int = (int64_t) value_num;
+          clamped = (uint8_t) tmp_int;
+
+          if (clamped + 0.5 < value_num
+              || (clamped + 0.5 == value_num && clamped % 2 == 1))
+          {
+            clamped ++;
+          }
+        }
+        *((uint8_t *) target_p) = clamped;
+        break;
+      }
+      case LIT_MAGIC_STRING_INT16_ARRAY_UL:
+      {
+        TYPEDARRAY_SET_INDEXED_ELEMENT (int16_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_UINT16_ARRAY_UL:
+      {
+        TYPEDARRAY_SET_INDEXED_ELEMENT (uint16_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_INT32_ARRAY_UL:
+      {
+        TYPEDARRAY_SET_INDEXED_ELEMENT (int32_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_UINT32_ARRAY_UL:
+      {
+        TYPEDARRAY_SET_INDEXED_ELEMENT (uint32_t);
+        break;
+      }
+      case LIT_MAGIC_STRING_FLOAT32_ARRAY_UL:
+      {
+        *((float *) target_p) = (float) value_num;
+        break;
+      }
+      case LIT_MAGIC_STRING_FLOAT64_ARRAY_UL:
+      {
+        *((double *) target_p) = (double) value_num;
```

Comment:
This seems a lot of code duplication, perhaps we should combine them. Code size matters :)